app/controllers/orders_controller.rb

```diff
@@ -1,48 +1,75 @@
 class OrdersController < ApplicationController
-  before_action :find_order, only: [:cart, :checkout, :update, :receipt]
-  before_action :check_access, only: [:cart, :checkout, :receipt]
-
+  before_action :set_order, only: [:cart, :checkout, :add_to_cart, :update, :receipt]
```

Comment:
It seems like it would be more effective to use the `except` on this rather than the `only`